overlord/snapstate/snapstate_test.go

```diff
@@ -5347,6 +5347,68 @@ func (s *snapmgrTestSuite) TestConflictExclusive(c *C) {
 	c.Check(err, ErrorMatches, `other changes in progress \(conflicting change "install-snap-a"\), change "create-recovery-system" not allowed until they are done`)
 }
 
+type noopBackend struct{}
+
+func (d noopBackend) Checkpoint(_ []byte) error { return nil }
+
+func (d noopBackend) EnsureBefore(_ time.Duration) {}
+
+func (d noopBackend) RequestRestart(_ state.RestartType) {}
+
+func (s *snapmgrTestSuite) TestOnlyConflictOnRelevantRunnableTask(c *C) {
+	// statuses that have no future work to be done
+	stoppedStatuses := []state.Status{state.ErrorStatus, state.HoldStatus, state.DoneStatus, state.UndoneStatus}
+
+	for id := 0; id < state.NStatuses; id++ {
+		status := state.Status(id)
+
+		s.state = state.New(noopBackend{})
```

Comment:
You can just use `state.New(nil)`, no need to have a dummy noopBackend.

Also a nitpick^2, maybe just introduce a local state variable, e.g. `st := state.New(nil)`.

Reply:
Thanks, I applied those suggestions